test/ecto/adapters/sqlite3/connection_test.exs

```diff
@@ -2414,6 +2414,28 @@ defmodule Ecto.Adapters.SQLite3.ConnectionTest do
     assert execute_ddl(integer) == [~s/CREATE TABLE "posts" ("id" INTEGER PRIMARY KEY)/]
   end
 
+  describe "to_constraints/2" do
+    alias Ecto.Adapters.SQLite3.Connection
+
+    test "simple unique index" do
+      error = %Exqlite.Error{message: "UNIQUE constraint failed: one.two.three"}
+      assert Connection.to_constraints(error, []) == [unique: "one_two_three_index"]
```

Comment:
Shouldn't this be `unique: "one.two.three"`?

Reply:
No, not in my testing.
I've done some more testing and added specific index examples I've used so it can be reproduced / tested against future SQLite versions.

Does it make more sense now?
Thanks for looking at this.